marketplace-core/src/main/java/org/pentaho/marketplace/endpoints/MarketplaceService.java

```diff
@@ -75,7 +76,7 @@ public StringOperationResultDTO hello() {
   @GET
   @Path( "/plugins" )
   @Produces( MediaType.APPLICATION_JSON )
-  public IterablePluginOperationResultDTO getPlugins() {
+  public Response getPlugins() {
```

Comment:
I see your point and both approaches have pros and cons. For the time being add the javax.servlet dependency in order to keep the return type explicit.